src/Schema/SchemaValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flagbit\Bundle\CategoryBundle\Schema;
+
+use JsonSchema\Validator;
+use LogicException;
+
+use function json_decode;
+use function json_encode;
+use function realpath;
+use function sprintf;
+
+use const JSON_THROW_ON_ERROR;
+
+/**
+ * @internal
+ */
+class SchemaValidator
+{
+    private object $schema;
+
+    public function __construct(string $schemaPath)
+    {
+        $realpath = realpath(sprintf('%s/%s', __DIR__, $schemaPath));
+        if ($realpath === false) {
+            throw new LogicException(sprintf('File %s not found', $schemaPath));
```

Comment:
`LogicException` is very generic, maybe consider using or creating a more specific exception type.

Reply:
I've added new Exception classes according to the coding style.